File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<char,int> hash;
        
        for(int i = 0; i < nums.size(); i++) {
            hash[nums[i]]++;
            
            if(hash[nums[i]] > nums.size()/2) {
                return nums[i];
            }
        }
        
        return NULL;
    }
};

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
      map = {}
      majority = len(nums) // 2
      for element in nums:
        if element not in map:
          map[element] = 1
        else:
          map[element] += 1
         # If an element appears more than n/2 times, return that element
        if map[element] > majority:
          return element

      return None